### PR TITLE
Chore: Updated plugin schema docs ref PR@38463

### PR DESCRIPTION
**Pull request by srclosson**
**What this PR does / why we need it**: See https://github.com/grafana/grafana/pull/38463. This updates the metadata.md file as a result of the updated schema docs.

**Which issue(s) this PR fixes**: This is in addition to PR https://github.com/grafana/grafana/pull/38463

### CODE REVIEW

**Review by marcusolsson — APPROVED** [review]

File: docs/sources/developers/plugins/metadata.md [hunk 1]
```diff
@@ -13,32 +13,33 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
+| `type`               | string                        | **Yes**  | Plugin type. Possible values are: `app`, `datasource`, `panel`.                                                                                                                                                                                                                                                                                                                                         |
+| `$schema`            | string                        | No       | Schema definition for the plugin.json file                                                                                                                                                                                                                                                                                                                                                              |
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/metadata.md:23]:
```suggestion
| `$schema`            | string                        | No       | Schema definition for the plugin.json file.                                                                                                                                                                                                                                                                                                                                                              |
```

File: docs/sources/developers/plugins/metadata.md [hunk 2]
```diff
@@ -13,32 +13,33 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
+| `alerting`           | boolean                       | No       | For data source plugins. If the plugin supports alerting.                                                                                                                                                                                                                                                                                                                                               |
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/metadata.md:24]:
```suggestion
| `alerting`           | boolean                       | No       | For data source plugins, if the plugin supports alerting.                                                                                                                                                                                                                                                                                                                                               |
```

File: docs/sources/developers/plugins/metadata.md [hunk 3]
```diff
@@ -13,32 +13,33 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
+| `annotations`        | boolean                       | No       | For data source plugins. If the plugin supports annotation queries.                                                                                                                                                                                                                                                                                                                                     |
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/metadata.md:25]:
```suggestion
| `annotations`        | boolean                       | No       | For data source plugins, if the plugin supports annotation queries.                                                                                                                                                                                                                                                                                                                                     |
```

File: docs/sources/developers/plugins/metadata.md [hunk 4]
```diff
@@ -13,32 +13,33 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
+| `category`           | string                        | No       | Plugin category used on the Add data source page. Possible values are: `tsdb`, `logging`, `cloud`, `tracing`, `sql`, `enterprise`, `other`.                                                                                                                                                                                                                                                             |
+| `enterpriseFeatures` | [object](#enterprisefeatures) | No       | Grafana Enerprise specific features                                                                                                                                                                                                                                                                                                                                                                     |
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/metadata.md:29]:
```suggestion
| `enterpriseFeatures` | [object](#enterprisefeatures) | No       | Grafana Enerprise specific features.                                                                                                                                                                                                                                                                                                                                                                     |
```

File: docs/sources/developers/plugins/metadata.md [hunk 5]
```diff
@@ -13,32 +13,33 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
+| `executable`         | string                        | No       | The first part of the file name of the backend component executable. There can be multiple executables built for different operating system and architecture. Grafana will check for executables named `<executable>_<$GOOS>_<lower case $GOARCH><.exe for Windows>`, e.g. `plugin_linux_amd64`. Combination of $GOOS and $GOARCH can be found here: https://golang.org/doc/install/source#environment. |
+| `hiddenQueries`      | boolean                       | No       |                                                                                                                                                                                                                                                                                                                                                                                                         |
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/metadata.md:31]: Need description.

File: docs/sources/developers/plugins/metadata.md [hunk 6]
```diff
@@ -13,32 +13,33 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
+| `logs`               | boolean                       | No       | For data source plugins. If the plugin supports logs.                                                                                                                                                                                                                                                                                                                                                   |
+| `metrics`            | boolean                       | No       | For data source plugins. If the plugin supports metric queries. Used in the Explore feature.                                                                                                                                                                                                                                                                                                            |
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/metadata.md:34]:
```suggestion
| `metrics`            | boolean                       | No       | For data source plugins, if the plugin supports metric queries. Used in Explore.                                                                                                                                                                                                                                                                                                            |
```

File: docs/sources/developers/plugins/metadata.md [hunk 7]
```diff
@@ -13,32 +13,33 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
+| `streaming`          | boolean                       | No       | For data source plugins. If the plugin supports streaming.                                                                                                                                                                                                                                                                                                                                              |
+| `tables`             | boolean                       | No       |                                                                                                                                                                                                                                                                                                                                                                                                         |
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/metadata.md:41]: Need description.

File: docs/sources/developers/plugins/metadata.md [hunk 8]
```diff
@@ -65,6 +66,16 @@ Plugin dependency. Used to display information about plugin dependencies in the
...
 
+## enterpriseFeatures
+
+Grafana Enerprise specific features
+
+### Properties
+
+| Property                  | Type    | Required | Description                                                        |
+| ------------------------- | ------- | -------- | ------------------------------------------------------------------ |
+| `healthDiagnosticsErrors` | boolean | No       | Enable/Disable health diagnostics errors. Requires Grafana >=7.5.5 |
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/metadata.md:77]:
```suggestion
| `healthDiagnosticsErrors` | boolean | No       | Enable/Disable health diagnostics errors. Requires Grafana >=7.5.5. |
```

File: docs/sources/developers/plugins/metadata.md [hunk 9]
```diff
@@ -13,32 +13,33 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
+| `tables`             | boolean                       | No       |                                                                                                                                                                                                                                                                                                                                                                                                         |
+| `tracing`            | boolean                       | No       | For data source plugins. If the plugin supports tracing.                                                                                                                                                                                                                                                                                                                                                |
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/metadata.md:42]:
```suggestion
| `tracing`            | boolean                       | No       | For data source plugins, if the plugin supports tracing.                                                                                                                                                                                                                                                                                                                                                |
```

File: docs/sources/developers/plugins/metadata.md [hunk 10]
```diff
@@ -13,32 +13,33 @@ The plugin.json file is required for all plugins. When Grafana starts, it scans
...
+| `state`              | string                        | No       | Marks a plugin as a pre-release. Possible values are: `alpha`, `beta`.                                                                                                                                                                                                                                                                                                                                  |
+| `streaming`          | boolean                       | No       | For data source plugins. If the plugin supports streaming.                                                                                                                                                                                                                                                                                                                                              |
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/metadata.md:40]:
```suggestion
| `streaming`          | boolean                       | No       | For data source plugins, if the plugin supports streaming.                                                                                                                                                                                                                                                                                                                                              |
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added copy edit suggestions.

**Review by achatterjee-grafana — APPROVED** [review]
LGTM

**Pull request merged**